File: Backend/user-service/spring_boot_backend_template/spring_boot_backend_template/src/main/java/com/backend/entity/Seat.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "seats")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Seat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String seatNumber;
    private boolean booked;

    @ManyToOne
    @JoinColumn(name = "bus_id")
    private Bus bus;
}

File: Backend/admin-service/admin/admin-service/Models/Seat.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace admin_service.Models
{
    public class Seat
    {
        public long Id { get; set; }

        [Column("seat_number")]
        public string SeatNumber { get; set; }

        [Column("bus_id")]
        public long BusId { get; set; }

        public Bus Bus { get; set; }
    }
}
